llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -1210,8 +1210,16 @@ bool SPIRVInstructionSelector::selectUnOp(Register ResVReg,
     for (MachineRegisterInfo::def_instr_iterator DefIt =
              MRI->def_instr_begin(SrcReg);
          DefIt != MRI->def_instr_end(); DefIt = std::next(DefIt)) {
-      if ((*DefIt).getOpcode() == TargetOpcode::G_GLOBAL_VALUE ||
-          (*DefIt).getOpcode() == SPIRV::OpVariable) {
+      unsigned DefOpCode = (*DefIt).getOpcode();
+      if (DefOpCode == SPIRV::ASSIGN_TYPE) {
```

Comment:
Isn't this loop only iterating on register definitions and not uses?
I thought `spv.assign.type` as `spv.assign.ptr.type` were intrinsics registering as having a side-effect, but returning no values?